File: src/main/java/xyz/gnarbot/gnar/Credentials.java
package xyz.gnarbot.gnar;

import org.json.JSONObject;
import org.json.JSONTokener;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;

public class Credentials {
    public final static String PRODUCTION;
    public final static String BETA;

    public final static String MARVEL_PU;
    public final static String MARVEL_PR;

    public final static String ABAL_URL;
    public final static String ABAL_TOKEN;

    public final static String LEAUGE;
    public final static String IMGFLIP;

    public final static String CARBONITEX;

    public final static String MASHAPE;

    static {
        JSONObject jso = null;
        try {
            File file = new File(new File("data"), "credentials.json");
            jso = new JSONObject(new JSONTokener(new FileReader(file)));
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }

        assert jso != null;
        JSONObject token = jso.getJSONObject("token");
        PRODUCTION = token.optString("production");
        BETA = token.optString("beta");
        JSONObject marvel = jso.optJSONObject("marvel");
        MARVEL_PU = marvel.optString("pu");
        MARVEL_PR = marvel.optString("pr");
        JSONObject abal = jso.optJSONObject("abal");
        ABAL_URL = abal.optString("url");
        ABAL_TOKEN = abal.optString("token");
        LEAUGE = jso.optString("leauge");
        IMGFLIP = jso.optString("imgflip");
        CARBONITEX = jso.optString("carbonitex");
        MASHAPE = jso.optString("mashape");
    }

    private Credentials() {}
}

File: src/main/kotlin/xyz/gnarbot/gnar/Credentials.kt
package xyz.gnarbot.gnar

import org.json.JSONObject
import org.json.JSONTokener
import java.io.File
import java.io.FileReader
import kotlin.jvm.JvmStatic as static

object Credentials {
    private val file = File(Bot.files.data, "credentials.json")
    private val jso = JSONObject(JSONTokener(FileReader(file)))

    private val token = jso.getJSONObject("token")
    @JvmField val PRODUCTION: String = token.optString("production") ?: ""
    @JvmField val BETA: String = token.optString("beta") ?: ""

    private val marvel = jso.optJSONObject("marvel")
    @JvmField val MARVEL_PU: String = marvel.optString("pu") ?: ""
    @JvmField val MARVEL_PR: String = marvel.optString("pr") ?: ""

    private val abal = jso.optJSONObject("abal")
    @JvmField val ABAL_URL: String = abal.optString("url") ?: ""
    @JvmField val ABAL_TOKEN: String = abal.optString("token") ?: ""

    @JvmField val LEAUGE: String = jso.optString("leauge") ?: ""
    @JvmField val IMGFLIP: String = jso.optString("imgflip") ?: ""

    @JvmField val CARBONITEX: String = jso.optString("carbonitex") ?: ""

    @JvmField val MASHAPE: String = jso.optString("mashape") ?: ""
}